extensions/asde-deployment/package.json

```diff
@@ -335,7 +335,7 @@
                         {
                           "label": "%vm_password%",
                           "variableName": "AZDATA_NB_VAR_ASDE_VM_PASSWORD",
-                          "type": "password",
+                          "type": "azurevm_password",
```

Comment:
Why not just plumb through the input regex validation we already have set up? It would be super grand not to have to keep making new types like this for every different type of password we have.

Reply:
I have been hard at work to avoid the proliferation of different 'types' for fields. If two field types differ only slightly, they should just be variants of each other. Here as Charles suggests just adding the right validations will hopefully do the trick. I know, currently, only text input type allows for the regex validation that he mentions but the validation work that I have ongoing with open PR addresses that and you will be able to define a custom validation even on the password input field type.